x/genutil/client/cli/validate_genesis.go

```diff
@@ -54,3 +51,19 @@ func ValidateGenesisCmd(mbm module.BasicManager, txEncCfg client.TxEncodingConfi
 		},
 	}
 }
+
+// validateGenDoc reads a genesis file and validates that it is a correct
+// Tendermint GenesisDoc. This function does not do any cosmos-related
+// validation.
+func validateGenDoc(importGenesisFile string) (*tmtypes.GenesisDoc, error) {
+	genDoc, err := tmtypes.GenesisDocFromFile(importGenesisFile)
+	if err != nil {
+		return nil, fmt.Errorf("%s. Make sure that"+
+			" you have correctly migrated all Tendermint consensus params, please see the"+
+			" chain migration guide at %s for more info",
+			err.Error(), "https://docs.cosmos.network/master/migrations/chain-upgrade-guide-040.html",
```

Comment:
This page still doesn't exist, I'll try to make that happen this week #7771. Are we still okay to merge this PR first?